### PR TITLE
explain why type is at the end of packet

### LINKED ISSUE #58: Clarify MessageType position in Section 2.2.2
2) s2.2.2, #1: I gotta ask why the MessageType byte isn’t first? Wouldn’t putting the MessageType 
first allow you avoid buffering everything before proceeding?  It seems like you’re counting on 
nobody sending a string of zeros as the first byte of the EKT field to mess with implementations.

### PR DESCRIPTION
**Pull request by fluffy**
Fixes #58

### CODE REVIEW

File: srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml [hunk 1]
```diff
@@ -384,7 +384,18 @@ EKTField = FullEKTField / ShortEKTField / ExtensionEKTField
               ShortEKTField, the ShortEKTField is removed from the
               packet then normal SRTP or SRTCP processing occurs. If
               the packet contains a FullEKTField, then processing
-              continues as described below.</t>
+              continues as described below.
+              Using the last byte of the packet to indicate the type
+              is an unusual design however, the reason it is this way,
```
Comment (paulej) [inline srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml]: Would suggest changing this line to be:

```
is an unusual design; however, the reason it is this way
```

**Comment by fluffy** [comment]
Does this way of phrasing it look better ?

File: srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml [hunk 2]
```diff
@@ -384,7 +384,17 @@ EKTField = FullEKTField / ShortEKTField / ExtensionEKTField
...
-              continues as described below.</t>
+              continues as described below.
+              The reason for using the last byte of the packet to
+              indicate the type is that the length of the SRTP or
+              SRTCP part is not known until the decryption has
+              occurred. At this point in the processing, there is no
+              easy way to know where the EKT field would
+              start. However, the whole UDP packet has been received
+              so instead of the starting at the front of the packet,
+              the parsing work backwards off the end of the packet and
```
Comment (paulej) [inline srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml]: "work" => "works"

File: srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml [hunk 3]
```diff
@@ -384,7 +384,17 @@ EKTField = FullEKTField / ShortEKTField / ExtensionEKTField
...
+              thus the type is put at the very end of the packet.
```
Comment (paulej) [inline srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml]: The type doesn't indicate length, though.  The length does.

The type just tells us whether it's a short tag or long tag (or some TBD tag).  The length precedes the type in the long tag.

So, the type is there to tell us if we have a short tag (one octet) or another tag type.  In the latter case, an EKT tag length is introduced  just before the type field.

**Pull request merged**